Add files via upload

FranciscoJLZ opened the pull request:
Entregable, fecha 01/10/2024. Trabajo de herencia, se agregó la opcion de incorporar camiones a la lista del registro del automotor, la opcion de motos ya se agregó la semana pasada

FranciscoJLZ committed:
Add files via upload

SofiZ2001 commented on reCamiones.ts:
```diff
@@ -0,0 +1,35 @@
+        }
+    }
+
+}
```
Creo que esta rara esta herencia.
Deberias tener una clase padre Vehiculo que tenga como hijos Auto, Camion y Moto.
Todo lo que veas en comun entre esas 3 clases hijas debería tenerlas las clase padre

Ej. todas las motos, autos y camiones tienen patente, modelo, marca, y valor. Para evitar repetir codigo en esas clases, directamente lo tiene la clase padre Vehiculo, y despues las hijas heredan esa funcionalidad

FranciscoJLZ committed:
entregableHerenciaCorreciones

Se agregó la clase padre Vehiculos, que hereda a las clases: Camiones,camionetas,motos y autos. Ademas se creó la clase patentes, la cual le asigna a los nuevos agregados una patente.

FranciscoJLZ committed:
Merge branch 'main' into entregableHerencia

SofiZ2001 commented on reMain.ts:
```diff
@@ -6,15 +6,19 @@ import  { Autos } from "./reAutos"; import { Camionetas } from "./reCamionetas";
+
+export class RegistroAutomotor{
+  
     private vehiculos: Autos[] | Motos[] | Camionetas[] = [];
```
Este arreglo de vehiculos ahora con herencia aplicada podría ser solamente de tipo Vehiculo

SofiZ2001 commented on reMain.ts:
```diff
@@ -6,15 +6,19 @@ import  { Autos } from "./reAutos"; import { Camionetas } from "./reCamionetas";
         this.vehiculos=this.vehiculos;
     }
-    //-------------
+
```
RegistroAutomotor podría tener las funciones de agregar, eliminar y modificar vehiculos

SofiZ2001 reviewed (changes requested):
2 pequeños comments fran

FranciscoJLZ committed:
Se agregaron las correciones de los metodos

SofiZ2001 commented on reMain.ts:
```diff
@@ -1,20 +1,38 @@
-export class RegistroAutomotor {
-    private vehiculos: Autos[] | Motos[] | Camionetas[] = [];
+
+class RegistroAutomotor extends Vehiculos{
```
RegistroAutomotor **NO ES** un Vehiculo (herencia), por lo que no debería extender de esa clase.
RegistroAutomotor **TIENE** vehiculos (composicion), entonces debería tener una lista de Vehiculos

SofiZ2001 commented on reMotos.ts:
```diff
@@ -1,57 +1,7 @@
-            console.log("Iterador inválido.");
-        }
+    constructor(marca:string,modelo:string,valor:string){
+        super(marca,modelo,valor)
```
Para no dejar la clase vacía, agregate una funcion unica de la clase y que se diferencie de las demas

SofiZ2001 commented on reCamionetas.ts:
```diff
@@ -1,57 +1,7 @@
-            console.log("Iterador inválido.");
-        }
+    constructor(marca: string, modelo: string, valor: string){
+     super(marca,modelo,valor);   
```
Para no dejar la clase vacía, agregate una funcion unica de la clase y que se diferencie de las demas

SofiZ2001 commented on reAutos.ts:
```diff
@@ -1,58 +1,8 @@
-            console.log("Iterador inválido.");
-        }
+    constructor(marca: string, modelo: string, valor: string){
+        super(marca,modelo,valor)
```
Para no dejar la clase vacía, agregate una funcion unica de la clase y que se diferencie de las demas

SofiZ2001 commented on patentes.ts:
```diff
@@ -0,0 +1,33 @@
+  //setters--------
+  setPatente(patente: string) {
+    this.patente = patente;
+  }
```
Recorda los modificadores de acceso de las funciones

SofiZ2001 reviewed (changes requested):
Otros comments fran

FranciscoJLZ committed:
se agrego una func unica a cada clase y dms cambis

SofiZ2001 commented on patentes.ts:
```diff
@@ -0,0 +1,33 @@
+
+//     return patente;
+//   }
+// }
```
Si no lo vas a utilizar borralo

SofiZ2001 commented on reMotos.ts:
```diff
@@ -1,57 +1,26 @@
-    private modeloMoto: string;
-    private valorMoto: string;
+import { RegistroAutomotor } from "./reMain"
+export class Motos extends RegistroAutomotor {
```
No esta bien esta herencia, debería heredar de Vehiculo

SofiZ2001 commented on reMotos.ts:
```diff
@@ -1,57 +1,26 @@
+    }
+    //setters----
+    setCilindrada(cilindrada: number) {
+        return this.cilindrada;
```
modificador de acceso

SofiZ2001 commented on reMain.ts:
```diff
@@ -1,20 +1,81 @@
+    setModelo(modelo: string): void {
+        this.modelo = modelo;
+    }
+    setValor(valor:string): void{
```
modificador de acceso

SofiZ2001 commented on reCamionetas.ts:
```diff
@@ -1,57 +1,26 @@
-    private modeloCamioneta: string;
-    private valorCamioneta: string;
+import { RegistroAutomotor } from "./reMain";
+export class Camionetas extends RegistroAutomotor {
```
No esta bien esta herencia, debería heredar de Vehiculo

SofiZ2001 commented on reCamiones.ts:
```diff
@@ -0,0 +1,25 @@
+import { RegistroAutomotor } from "./reMain";
+export class Camiones extends RegistroAutomotor{
```
No esta bien esta herencia, debería heredar de Vehiculo

SofiZ2001 commented on reAutos.ts:
```diff
@@ -1,58 +1,27 @@
-    private modeloAuto: string;
-    private valorAuto: string;
+import { RegistroAutomotor } from "./reMain"
+export class Autos extends RegistroAutomotor {
```
No esta bien esta herencia, debería heredar de Vehiculo

SofiZ2001 reviewed (changes requested):
comments

FranciscoJLZ committed:
Se razron. cambios en main y se reinc. clase Vehi.

FranciscoJLZ committed:
se realizaron cambios en main e incorporo vehiculo

FranciscoJLZ committed:
Merge branch 'entregableHerencia' of https://github.com/FranciscoJLZ/entregableOlavarria into entregableHerencia

SofiZ2001 reviewed (changes requested):
Te quedo un comment pendiente

FranciscoJLZ committed:
se eliminó patentes.ts

SofiZ2001 reviewed (approved):
LG2M
